modules/dnn/perf/perf_convolution.cpp

```diff
@@ -773,16 +774,14 @@ struct ConvParamGenerator
 
     ::testing::internal::ParamGenerator<ConvParam_t> all() const
     {
-#if 1
-        // default to 20 for each type of convolution
-        const int NUM = 20;
-#else
-        const int NUM = size;
-#endif
-
-        ConvParam_t v_[NUM];
+        int NUM = size;
+        static size_t DNN_LIMIT_CONV = utils::getConfigurationParameterSizeT("OPENCV_TEST_DNN_LIMIT_CONV", 0);
+        if (DNN_LIMIT_CONV > 0)
+            NUM = std::min(NUM, (int)DNN_LIMIT_CONV);
+
+        std::vector<ConvParam_t> v_(NUM);
```

Comment:
The PR changes default behaviour. All cases executed by default, but not only the first 20. Is it expected behaviour?